### PR TITLE
Remove default backends

### LINKED ISSUE #64: Unit tests do not use test backends
View typedefs are used throughout tests instead of Views with intended backends (TEST_MEMSPACE) - only default space tests are run

- [x] Cell capture (#121)
- [x] Nucleation (#213)
- [x] Cell data (#215)
- [x] Temperature (#223)
- [x] Communication, Other (free functions, etc.)

### PR DESCRIPTION
**Pull request by streeve**
Remove all typedefs for default View types. `CAtypes` now only defines cell types

Any large codeblock changes come from templating (moving from cpp to hpp)

Closes #64 
Follow-on to #253 (a bit of this really should have been done there..)

### CODE REVIEW

**Comment by MattRolchigo** [comment]
It looks like you use lowercase/underscores for local type definitions or memory spaces, like `view_type_double_host` in the declarations of the hpp files, and then capital letters when there's a function templated on a typedef, like `template <typename ViewTypeInt3dHost>
    RepresentativeRegion(nlohmann::json AnalysisData, ..., ViewTypeInt3dHost GrainID) {`. Is that by design?

**Comment by streeve** [comment]
> It looks like you use lowercase/underscores for local type definitions or memory spaces, like `view_type_double_host` in the declarations of the hpp files, and then capital letters when there's a function templated on a typedef, like `template <typename ViewTypeInt3dHost> RepresentativeRegion(nlohmann::json AnalysisData, ..., ViewTypeInt3dHost GrainID) {`. Is that by design?

That's definitely a personal habit (style from other projects). At this point it's all over the code

**Comment by MattRolchigo** [comment]
Good to know, I don't have an opinion on it either way but I just wanted to make sure I understood the formatting. Do you want me to wait to work on the last struct (Orientation, which will hold things like GrainUnitVector, getGrainNumber, NGrainOrientations) until this is done to avoid massive merge conflicts?

**Comment by streeve** [comment]
> Good to know, I don't have an opinion on it either way but I just wanted to make sure I understood the formatting. Do you want me to wait to work on the last struct (Orientation, which will hold things like GrainUnitVector, getGrainNumber, NGrainOrientations) until this is done to avoid massive merge conflicts?

I think this is done..

**Comment by MattRolchigo** [comment]
Let me test it for a few problems on Summit/Crusher to make sure it's working and then I'll approve

File: src/CAprint.hpp [hunk 1]
```diff
@@ -173,17 +175,18 @@ struct Print {
                 printViewData(id, Grainplot, grid, grid.nz_layer, "short", "LayerID", LayerID_WholeDomain);
             }
             if (_inputs.PrintInitMeltTimeStep) {
-                ViewI MeltTimeStep = temperature.extract_tm_tl_cr_data<ViewI>(0, grid.domain_size);
+                auto MeltTimeStep = temperature.template extract_tm_tl_cr_data<view_int>(0, grid.domain_size);
```
Comment (MattRolchigo) [inline src/CAprint.hpp:178]: What's the word "template" do here? It didn't seem like you changed extract_tm_tl_cr_data at all, what made this change necessary?

**Review by MattRolchigo — APPROVED** [review]

**Pull request merged**